### DISCUSSION TITLE
Initial data "effects" don't run the emit events

### DISCUSSION THREAD

**Discussion by timrspratt** [livewire/livewire#3081] [7. Bugs (archived)]
### Description
I have a use case where a nested component triggers `$this->emitUp()` within its `mount` method, and this data is included in the `wire:initial-data` (as `effects.emits`), yet it appears not to be triggered in the `Component` constructor `js/component/index.js` when the component is processed by the browser.

**Context**
- Livewire version: 2.2.2
- Laravel version: 8.1.0
- Browser: Chrome

**Comment by timrspratt** [comment]
I've managed to use `wire:init` for my use case, but leaving this open for your consideration incase it would still be counted as a bug. Thanks

**Comment by calebporzio** [comment]
Thanks for reporting this.

Yeah, `$this->emit` calls in `mount` SHOULD be received by other components on the page on-load.

**Comment by isaackearl** [comment]
This bug has affected my team recently. I made another ticket that was closed because I didn't realize this already existed.  Here is the simple example I created there


```
class ComponentOne extends Component
{

    // emitting an event to be picked up by the other component
    public function mount(){
      $this->emit('hello');
    }

    public function render()
    {
        return view('component-one');
    }
}
```
second component
```
class ComponentTwo extends Component
{

    protected $listeners = ['hello' => 'doSomething'];    

    public function doSomething(){
        dump('this never happens');
    }

    public function render()
    {
        return view('component-two');
    }
}
```

In this scenario even though both components are rendered on the same page, the event that is emitted from componentOne doesn't do anything.

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)

**Comment by isaackearl** [comment]
REOPEN

**Comment by isaackearl** [comment]
I went ahead and said re-open because I believe this is a legitimate issue even though there are some workarounds.  I plan on trying to create a dusk failing test eventually, but unfortunately haven't had the time.

**Comment by MACscr** [comment]
I am having this issue as well. Would love to see it working. I am finding a lot of use cases where emits aren't working as someone would expect nor documented that they won't as well and that leads to a lot of frustration.

**Comment by whoisthisstud** [comment]
I'm having this same issue. I have an emit in the mount() method and a script listening for the event to dump to console and it never reaches the script.

````
public function mount(?Quote $quote) {
    ....
    $this->emit('urlChange', [ 'url' => route('quotes.index',['quote'=> $this->quote->uuid]) ]);
}
````
and in blade template
````
@push('stackedScripts')
<script type="text/javascript">
    window.livewire.on('urlChange', url => {
        console.log(url); // <- NEVER REACHES HERE
        // history.pushState(null, null, url);
    })
</script>
@endpush
````
